recipe/meta.yaml

```diff
@@ -12,6 +12,7 @@ requirements:
   run:
     - r-base
     - r-recommended
+    - mscorefonts
```

Comment:
hm...  mscorefonts is only available for linux64 according to https://anaconda.org/conda-forge/mscorefonts ... that would mean that the noarch:generic r package gets uninstallable on mac and windows.

Reply:
Or do I overlook something here?

Reply:
@dbast No, that is a very good point. Thanks for noting it! My main goal has been to fix the text in plots created in Docker containers and other minimal environments. I hadn't thought to check for macOS and Windows builds.

Reply:
Fonts should be noarch. Here's the recipe we've been using: https://github.com/AnacondaRecipes/aggregate/blob/master/fonts-anaconda-feedstock/recipe/meta.yaml

The font recipes are also in that repo.

Reply:
@mingwandroid That recipe looks super useful. Should we migrate it to conda-forge? There has been a lot of effort recently by others to make R only depend on conda-forge packages:

https://github.com/conda-forge/r-base-feedstock/issues/83
https://github.com/conda-forge/r-base-feedstock/issues/64

Reply:
Feel free to adopt it. What about metapackage to abstract which endeavour the fonts come from?  So we'd have fonts-anaconda and fonts-conda-forge, and each have different fonts-conda-ecosystem packages to indirect them?

Overall, though, I. do think adding fonts to `r` isn't the right decision for various technical reasons. Only packages that depend upon (at least transitively) our `fontconfig` package will have any possibility of accessing these fonts, so therefore introducing a `run_exports` into the `fontconfig` package is the right thing to do here, then everything just works everywhere automatically. That is very much the reason we came up with `run_exports`.

Reply:
We are literally saying when we craft the recipe "the fontconfig package (whose primary concern is managing fonts on this system) is a dependency of this application, therefore we need to install the datafiles that are part of that overall font configuration".

Reply:
I agree, a metapackage will help this be much more flexible.

Reply:
Maybe we soon have a noarch mscorefonts package https://github.com/conda-forge/mscorefonts-feedstock/pull/3 

So the conda-forge version of `fonts-conda-ecosystem` pkg could redirect to mscorefonts (after the PR gets merged) or to `fonts-conda-forge` (after that is created out of fonts-anaconda).

Reply:
I would recommend conda-forge adopt (or discuss for adoption) the set I selected and put them all in the metapackage. I was pretty careful to pick fonts that covered a large range of scientific glyphs while keeping size concerns in mind (not that I think size concerns are relevant for our eco-system given we use hardlinks everywhere anyway!).

Reply:
> I do think adding fonts to r isn't the right decision for various technical reasons.

My goal is that a conda user that installs R inside a Docker container can create plots. I am agnostic to how we implement this.

> Only packages that depend upon (at least transitively) our fontconfig package will have any possibility of accessing these fonts, so therefore introducing a run_exports into the fontconfig package is the right thing to do here, then everything just works everywhere automatically. That is very much the reason we came up with run_exports.

So I need to add `r-base` to the `run_exports` section of the `fontconfig` recipe?

https://github.com/conda-forge/fontconfig-feedstock/blob/master/recipe/meta.yaml#L28

Reply:
Are you sure? Reading also the other threads, I am seeing the following steps:
1. Creation of the 2 meta-packages and 4 font-packages in conda-forge: `fonts-conda-ecosystem` pointing to `fonts-conda-forge` which depends on `font-ttf-{ubuntu, inconsolata, dejavu-sans-mono, source-code-pro}`.
2. Adding `fonts-conda-ecosystem` to `run_exports` of `fontconfig`: all packages that have `fontconfig` as `host` dependency then receive those fonts as `run` dependency.
3. Updating all font-consuming packages that do not yet depend on fontconfig or any font package (e.g. r-ggplot2): Either adding fontconfig to the `host` dependencies (which then implicitly adds `fonts-conda-ecosystem` to `run` dependencies) or adding directly `fonts-conda-ecosystem` to `run` dependencies... (so no font dependencies are added the `r-base` or `r` package). 
4. Adding some tests scripts (from the other threads) to e.g. r-ggplot2 to proof that the fonts are found and the right types of fonts are available.

Are those the right steps?

Reply:
Exactly @dbast. But let us get concensus that this is the best way.